src/main/scala/lms/transformation/distributed_tensor/distributedTensorUnary.scala

```diff
@@ -45,6 +63,30 @@ trait FixedSizeDistributedTensorUnaryTypeLess extends FixedSizeDistributedTensor
         (() => {
           Accumulate(gradMap(a), Negate(gradMap(s), anno), anno); ()
         }) +=: backwardNodes
+     
+    case Node(s, "tensor_invert", tt::Backend.Const(anno:Anno)::(a:Backend.Sym)::_, _) =>
+        implicit val pos = Adapter.oldSourceMap(s)
+        forwardNodes += node
+
+        (() => {
+          Accumulate(gradMap(a), Invert(gradMap(s), anno), anno); ()
```

Comment:
I think we need to fix the computation for backward pass, and actually some of them are a bit tricky.
for instance, the `invert` is like y = f(x) where f(x) = 1/x. Then if y is used to compute z by z = g(y), we can apply the chain-rule, such that we know dz/dx = dz/dy * dy/dx.
as a result, the operations in `backwardNodes` should be dz/dy * dy/dx. We know that `dz/dy` is gradMap(s). Then the computation needed is to multiply it with `dy/dx`, in this case, 1/x^2.

So I think the code here should be
val square = Mul(transform(a), transform(a), anno)
val grad = Div(gradMap(s), square, anno)
Accumulate(gradMap(a), grad, anno); ()

it is also possible for us to enclose these computations to a new node (maybe called InvertGrad)

Reply:
I have gone ahead and made your suggested changes